lib/ace/virtual_renderer.js

```diff
@@ -912,12 +912,13 @@ var VirtualRenderer = function(container, theme) {
     this.$autosize = function() {
         var height = this.session.getScreenLength() * this.lineHeight;
         var maxHeight = this.$maxLines * this.lineHeight;
-        var desiredHeight = Math.max(
-            (this.$minLines||1) * this.lineHeight,
-            Math.min(maxHeight, height)
+        var desiredHeight = Math.min(maxHeight, 
```

Comment:
changing the order of min and max here fixes https://github.com/ajaxorg/ace/issues/2693: infinite loop if minlines is larger than maxlines